### DISCUSSION TITLE
Free field in the parameters

### DISCUSSION THREAD

**Discussion by philippegu56** [RaspAP/raspap-webgui#1663] [Ideas]
Hi, 
I'd like to enable hostapd's macaddr_acl management. As RaspAP doesn't support it natively yet, I've added the parameters to the hostapd.conf file, and it works. But as the file is overwritten each time parameters are modified from the GUI, I have to re-enter them manually. 
For the time being, I've modified the hostapd.php file to force the parameters.
To get around this problem, it may be possible to add a free field in the parameters that lets you set certain options, in my case :
`macaddr_acl=1`
`accept_mac_file=/etc/hostapd/accept`

Thanks in advance
Phil

**Comment by philippegu56** [comment]
I made a slightly more flexible modification, adding a /etc/hostapd/hostapd.conf.users file containing my additional parameters. 
And I added this code at the end of the updateHostapdConfig function in /includes/includes/hostapd.php

```
exec('cat '. RASPI_HOSTAPD_CONFIG . '.users', $hostapdconfigusers);
   
foreach ($hostapdconfigusers as $hostapdconfigusersline) {
    if (strlen($hostapdconfigusersline) === 0) {
        continue;
    }
  
if ($hostapdconfigusersline[0] != "#") {
        $arrLine = explode("=", $hostapdconfigusersline);
        $config.= "$arrLine[0]=$arrLine[1];
    }
};
```

**Reply by billz** [reply]
Thanks for sharing your solution. hostapd has [a lot of settings](https://w1.fi/cgit/hostap/tree/hostapd/hostapd.conf) that aren't exposed by the UI (for good reason) so a drop-in config makes sense for advanced users.

**Reply by billz** [reply]
Implemented as #1666

**Reply by billz** [reply]
@philippegu56 if you'd find it useful, I'm glad to give you a complimentary upgrade to Insiders for this suggestion

**Reply by philippegu56** [reply]
Hello and thank you for the offer, which I gladly accept.
